develop 브랜치에 push 될때 배포되는 테스트용 별도의 aws ec2 서버 구축

okjunghyeon opened the pull request:
## ✨ 변경 사항 설명

<!-- 🔍 이 PR에서 변경된 내용을 간략하게 설명해주세요. -->
### 테스트용 서버 api.sulee.site 새로 배포함

실제 운영환경인 api.un1hub.site 와는 별개로
develop에 merge 될때마다 갱신되는 테스트를 위한 서버 
api.sulee.site를 새로 배포했습니다.

```
api.sulee.site (stg, 스테이징 서버)
- develop 브랜치로 push 될경우 해당 코드가 배포됩니다.

api.un1hub.site (prod, 운영 서버)
- main 브랜치로 push 될경우 해당 코드가 배포됩니다.
```

앞으로 개인 기능 개발 이후에 pr을 통해 develop에 push하게되면
api.sulee.site 테스트 서버의 배포환경에서도 정상 동작하는지 확인 부탁드립니다.
(api.sulee.site는 제 별도의 aws 계정으로 생성한 ec2 입니다 🥲)

---
### application.yml 기능 분리 및 주석 추가

application.yml 항목들이 직관적이지 못하고 유지보수 및 수정이 어렵다고 판단하여
총 5개의 항목으로 분리하여 정리하였습니다.
```
공통적으로 적용되어야하는 설정은 application.yml

개발환경에서 적용되어야 하는 설정은 application-dev.yml

테스트 서버에 적용되어야하는 설정은 application-stg.yml

운영 서버에 적용되어야하는 설정은 application-prod.yml

테스트코드에 적용되어야하는 설정은 application-test.yml
```
해당사항에 맞춰서 추가하거나 수정해주시면 될 것 같습니다.

---
### build.gradle 의존성 항목별로 분리
build.gradle을 항목별로 주석과 함께 구분하여 코드 가독성 향상시켰습니다.






## ✅ 체크리스트

<!-- 📝 완료된 항목은 [x]로 표시해주세요. -->

- [x] 🏗️ 코드가 **코딩 스타일 가이드**를 준수합니다.
- [ ] 🧪 변경 사항에 대한 **테스트가 추가**되었습니다.
- [x] ✅ 모든 **테스트가 정상적으로 통과**합니다.
- [x] 📚 필요한 **문서가 업데이트**되었습니다.

## 🎨 스크린샷 (UI 변경이 있는 경우)

<!-- 🖼️ 변경 전/후 스크린샷을 추가해주세요. -->

## 🛠️ 테스트 방법 (필요시)
![image](https://github.com/user-attachments/assets/4f8cfdef-badb-46c6-993b-f2ab873d5389)

jsc5023 committed:
refactor: rq 패키지 위치 변경

jsc5023 committed:
feat: refreshToken에 cookie(http Only) 추가

jsc5023 committed:
feat: swaggerUI 를 볼수있도록 추가

jsc5023 committed:
refactor: 줄간격 enter 삭제

jsc5023 committed:
feat: 이메일 인증 기능 추가 및 테스트

jsc5023 committed:
feat: 이메일 인증필요로 인한 initData 수정

jsc5023 committed:
refactor: initData 학생데이터 패스워드를 studentPw으로 변경

jsc5023 committed:
feat: 교직원 회원가입 승인 경우 추가

jsc5023 committed:
test: 교직원 회원가입시 승인여부를 미승인으로 변경, 로그인시 미승인 예외처리 추가

jsc5023 committed:
feat: 교직원 테스트 및 반복된 테스트를 편하게 하기 위한 Helper 추가

okjunghyeon committed:
refactor: 메인 배포 환경에 대한 terraform 설정을 infra\prod로 경로 변경

okjunghyeon committed:
Merge branch 'develop' of https://github.com/prgrms-web-devcourse-final-project/WEB4_5_GPT_BE into 옥정현/feat-34

okjunghyeon committed:
feat: develop 브랜치에 merge 될경우 테스트 해볼수 있는 aws ec2 dev 설정 추가

okjunghyeon committed:
feat: develop 브랜치 push 배포에 대한 workflows 추가

- 운영환경과 개발환경에 대한 github actions 배포 동작 분리

okjunghyeon committed:
feat: 각 workflows 별 name 변경

okjunghyeon committed:
feat: 소스코드 읽기 및 패키지 쓰기 권한 추가

okjunghyeon committed:
feat: github 환경에서 redis 설정 추가

okjunghyeon committed:
feat: testcontainers 적용

okjunghyeon committed:
fix : ProfessorSignupRequest 파일명이 변경 안되어있었음"

jsc5023 committed:
Merge branch 'develop' of https://github.com/prgrms-web-devcourse-final-project/WEB4_5_GPT_BE into 정성철/feat-28

okjunghyeon committed:
refactor : ProfessorSignUpRequest, StudentSignUpRequest 의 up 대문자로 변경

jsc5023 committed:
refactor: ProfessorSignupRequest -> ProfessorSignUpRequest

okjunghyeon committed:
refactor : develop 브랜치로 push 될때만 개발 서버가 재 배포되도록 수정

jsc5023 committed:
refactor: 띄어쓰기 오류 수정

SalinatedCoffee reviewed (approved):
LGTM, 고생하셨습니다! :+1:

okjunghyeon committed:
feat : 개발서버와 운영서버가 서로 다른 Dockerfile을 가지고 빌드할 수 있도록 분리

okjunghyeon committed:
feat : 운영서버에도 정상 동작하는지 테스트

okjunghyeon committed:
feat : testContainer를 제거하고 actions에 redis를 추가하여 정상 동작하는지 확인

okjunghyeon committed:
feat : Docker Buildx 설치 과정에 host 네트워크를 연결해주고 Dockerfile.dev 빌드 과정에서 해당 네트워크에 존재하는 redis에 접근 할 수 있도록 수정

okjunghyeon committed:
fix : 로컬 환경에서 정상적으로 테스트데이터가 등록되도록 수정

okjunghyeon committed:
feat: 개발, 운영환경 배포 initData Profile 수정

okjunghyeon committed:
feat: dev 환경에서 배포를 위해 이미지를 build할때는 test 하지 않도록 설정

jsc5023 committed:
refactor: 이메일 보내느 부분과, 레디스 이메일 인증코드 분리

okjunghyeon committed:
feat: 운영 환경 테스트를 위한 actions 동작 브랜치 추가

jsc5023 committed:
test: 통합 테스트시, 메일을 보내지 않도록 수정

okjunghyeon committed:
refactor: @RedisTestContainerConfig 제거

jsc5023 committed:
Merge branch 'develop' of https://github.com/prgrms-web-devcourse-final-project/WEB4_5_GPT_BE into 정성철/feat-28

okjunghyeon committed:
Merge branch 'develop' of https://github.com/prgrms-web-devcourse-final-project/WEB4_5_GPT_BE into 옥정현/feat-34

okjunghyeon committed:
Merge branch '정성철/feat-28' of https://github.com/prgrms-web-devcourse-final-project/WEB4_5_GPT_BE into 옥정현/feat-34

okjunghyeon committed:
refactor: application.dev, stg, prod 세가지로 구분하여 개발,테스트,운영 환경이 독립적으로 동작하도록 수정

- 배포 관련 workflows 파일도 prod(운영)와 stg(테스트)로 구분되도록 수정
- backend/Dockerfile 또한 prod와 stg로 구분되도록 수정
- infra 관련 패키지 명 또한 dev에서 stg로 수정하여 직관적이도록 구현

okjunghyeon committed:
Merge branch 'develop' of https://github.com/prgrms-web-devcourse-final-project/WEB4_5_GPT_BE into 옥정현/feat-34

okjunghyeon committed:
fix: 기본 yml 설정이 무조건 dev가 아니라 default일 때 dev가 되도록 수정

okjunghyeon committed:
fix: 이미지를 빌드할 때 active 설정이 제대로 적용되도록 순서 수정

okjunghyeon committed:
test: 테스트 환경 배포 테스트용 커밋

okjunghyeon committed:
test: secret.yml 최신화 및 .default 변경

okjunghyeon committed:
refactor: paths 도커파일 경로 수정

okjunghyeon committed:
refactor :  application-dev.yml 제거하고 기본 application.yml으로 통합 및 메인 서버 테스트

okjunghyeon committed:
fix : develop merge 후 코드상에 테이블 변경이 많아 기존 배포된 스키마와 충돌, ddl-auto를 create로 설정해 삭제하고자 함

okjunghyeon committed:
feat: 테스트 서버에도 배포 적용

okjunghyeon committed:
refactor: yml 및 build.gradle에 주석을 추가하여 코드 가독성 향상

TheGreatKang reviewed (approved):
👍
